com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JdtSourceLookUpProvider.java

```diff
@@ -181,6 +187,37 @@ public String getSourceFileURI(String fullyQualifiedName, String sourcePath) {
         return null;
     }
 
+    private static final Pattern JAVA_VERSION_PATTERN = Pattern.compile("version \"(.*)\"");
```

Comment:
Just FYI that the pattern **could** not cover some corner case (though we haven't figure out why), see https://github.com/AdoptOpenJDK/openjdk-build/issues/1610

Reply:
Thanks for the info. I plan to reuse the JDT logic to get the Java version, where it uses `release` file under the Java home to get the version. I assume it's more stable regarding the jdt user base.

```java
	public synchronized String readReleaseVersion(File javaHome) {

		String version = ""; //$NON-NLS-1$

		if (Files.notExists(Paths.get(javaHome.getAbsolutePath(), RELEASE_FILE))) {
			return version;
		}
		try (Stream<String> lines = Files.lines(Paths.get(javaHome.getAbsolutePath(), RELEASE_FILE), Charset.defaultCharset()).filter(s -> s.contains(JAVA_VERSION))) {
			Optional<String> hasVersion = lines.findFirst();
			if (hasVersion.isPresent()) {
				String line = hasVersion.get();
				version = line.substring(14, line.length() - 1); // length of JAVA_VERSION + 2 in JAVA_VERSION="9"
			}
		}
		catch (UncheckedIOException | IOException e) {
			LaunchingPlugin.log(e);
		}

		return version;

	}
```

Reply:
@CsCherrYY FYI

Reply:
@testforstephen `release` file looks great! I tried several, **one distribution of AdoptOpenJDK uses major version** while others use full version. But it's the same value with `java -version` output.

Zulu:
```
IMPLEMENTOR="Azul Systems, Inc."
IMPLEMENTOR_VERSION="Zulu14.28+21-CA"
JAVA_VERSION="14.0.1"
```

Adopt:
```
IMPLEMENTOR="AdoptOpenJDK"
IMPLEMENTOR_VERSION="AdoptOpenJDK"
JAVA_VERSION="14"    // <-------------HERE
```
```
IMPLEMENTOR="AdoptOpenJDK"
IMPLEMENTOR_VERSION="AdoptOpenJDK"
JAVA_VERSION="13.0.2"
```

Oracle:
```
BUILD_TYPE="commercial"
IMPLEMENTOR="Oracle Corporation"
JAVA_VERSION="12.0.2"
```

Reply:
YES. In real world, maybe 99% cases will contain `release` file. You could find the Java version from the release file first. If it doesn't exist, then fall back to `java -version` command.